teek/_widgets/treeview.py

```diff
@@ -0,0 +1,452 @@
+            self._fallbacks[option] = value
+            # Look for handler function(s)
+            handlers = (self._handlers[option] if option in self._handlers
+                        else self._handlers['*'])
+
+            if not isinstance(handlers, Iterable):
+                handlers = [handlers]
+
+            for handler in handlers:
+                handler(None, option, value)
+        except Exception:
+            pass
+
+    def _get(self, option):
+        try:
+            # Look for handler function
+            handler = (self._handlers[option]
+                       if option in self._handlers else self._handlers['*'])
+
+            if isinstance(handler, Iterable):
+                handler = handler[0]
+
+            return handler(self._types.get(option, str), option)
+        except Exception:
+            # If exception, return fallback value
+            return (self._fallbacks[option]
+                    if option in self._fallbacks else None)
+
+    def _list_options(self):
+        return set(list(self._types.keys()) + list(self._special.keys()))
+
+    def _check_option(self, option):
+        # Only check if master handler function is not defined
+        if '*' not in self._list_options():
+            super()._check_option(option)
+
+    def push(self):
+        """
+        Update all values (specified in fallback dict) on remote side
+        """
+        for option, value in self._fallbacks.items():
+            self._set(option, value)
+
+
+class TreeviewColumn:
+    """
+    Represents a column of a treeview widget or one which is ready to be added
+    to a treeview widget. A column specifies the title, content anchor and size
+    options.
+
+    If you create a :class:`TreeviewColumn` instance, it is not initially
+    connected to a :class:`Treeview`. This is automatically done when it is
+    appended to the *columns* list of the :class:`Treeview` instance like this:
+    ::
+
+        treeview.columns.append(teek.TreeviewColumn())
+
+    A column can be created without any arguments. Then the name of the column
+    is automatically generated. If the first argument is given, then this name
+    is taken.
+
+    For convenience, a column can also be created using...
+    ::
+
+        treeview.columns.append('title')
+
+    ...which is exactly the same as this:
+    ::
+
+        treeview.columns.append(teek.TreeviewColumn(title='title'))
+
+    There are never multiple :class:`TreeviewColumn` objects that represent the
+    same column.
+
+    .. attribute:: config
+
+        Similar to the ``config`` attribute that widgets have. The available
+        options can be seen at the ``column`` and ``heading`` command section
+        in :man:`ttk_treeview(3tk)`.
+    """
+
+    next_col_num = 0
+
+    @make_thread_safe
+    def __init__(self, name=None, command=None, **kwargs):
+        self._name = name or 'C%d' % TreeviewColumn.next_col_num
+        self._treeview = kwargs.pop('treeview', None)
```

Comment:
If you want `treeview` to be a keyword-only argument, it might be more correct to utilize the asterisk (`*`) "operator", as you can see in this example:

```python
def foo(a, *, b=None):
    return a * (b if b is not None else 0)

foo(1, b=2)  # returns 2
foo(1, 2)  # raises TypeError: foo() takes 1 positional argument but 2 were given
```

Reply:
Ah ok, I will change that! Thanks.